docs/content/guides/integrations/approaches-to-writing-integrations.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: "Approaches to writing integrations"
+---
+
+# Approaches to writing integrations
+
+There are many approaches to writing integrations in Dagster. The choice of approach depends on the specific requirements of the integration, the level of control needed, and the complexity of the external system being integrated. By reviewing the pros and cons of each approach, it is possible to make an informed decision on the best method for a specific use case. The following are typical approaches that align with Dagster's best practices.
+
+- Resource providers
+- Factory methods
+- Multi-Asset decorators
+- Pipes protocol
+
+## Resource providers
+
+One of the most fundamental features that can be implemented in an integration is a resource object to interface with an external service. For example, the `dagster-snowflake` integration provides a custom [SnowflakeResource](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py) that is a wrapper around the Snowflake `connector` object.
+
+### Pros
+
+- **Simple** Implementing a resource wrapper is often the first step in flushing out a fully-featured integration.
+- **Reusable** Resources are a core building block in the Dagster ecosystem, and allow one to re-use code across assets.
+
+### Cons
+
+- **Limited abstraction** While the resource can be re-used throughout the codebase, it does not provide any higher level abstraction to assets or jobs.
+
+### Guide
+
+<Note>A guide for writing a resource based integration is coming soon!</Note>
+
+## Factory methods
```

Comment:
Which integration do you have in mind? I don't think we have any builtin integrations are written this way (the good ones). If we're just talking about how a user can build their own integration in factory, I think that's fine but maybe we could move it down to a "Other patterns".

Stepping back, I think this guide will have two audience:
1. Partners / Contributors who want to build integrations that other users in the public would use
2. Individual organizations/users who want to integrate Dagster with their tech stack, specific to their needs.

To me, factory methods fit 2 but not 1 because it limits the composability. Therefore, I thought maybe we could move it to a separate section and keep the 1 with those other blessed patterns.

Reply:
lets address this as a follow up!